src/client/opamConfigCommand.ml

```diff
@@ -779,6 +779,20 @@ type ('var,'config) var_confset =
   }
 
 let set_var svar value conf =
+  let global_option =
+    List.exists (fun (name, _) -> String.equal svar name) OpamFile.Config.fields
+  in
+  let switch_option =
+    List.exists (fun (name, _) -> String.equal svar name)
+      OpamFile.Switch_config.fields
+  in
+  if global_option || switch_option then
+    (OpamConsole.msg
```

Comment:
Slightly concerned about compatibility - although it's almost guaranteed to be the case that setting, say, `jobs` with `opam var` was a mistake, there could be situations where this is correct. Can the question be guarded either for when we're in interactive mode or for when we're in the default or 2.2 CLI?

Reply:
Fresh thought, coming back to this - how about displaying a warning and advertising the command to revert to it (`opam var set FOO=`) but then setting it without any prompt. It's almost certainly a mistake on the user's part, so we tell them how to fix it, but we still do as we were told (if under protest!)